src/main/scala/com/advancedtelematic/campaigner/db/Repository.scala

```diff
@@ -213,7 +216,11 @@ protected class CampaignRepository()(implicit db: Database, ec: ExecutionContext
 
   def persist(campaign: Campaign, groups: Set[GroupId], metadata: Seq[CampaignMetadata]): Future[CampaignId] = db.run {
     val f = for {
-      _ <- (Schema.campaigns += campaign).handleIntegrityErrors(Errors.ConflictingCampaign)
+      _ <- (Schema.campaigns += campaign).asTry.flatMap {
```

Comment:
Also, is there a better way to do this? It feels hacky, but I wanted to differentiate and throw a different error when the cause is that there is no valid FK from campaigns to updates.